aas-gui/Frontend/aas-web-gui/src/components/SubmodelElements/Range.vue

```diff
@@ -80,11 +80,26 @@ export default defineComponent({
     },
 
     methods: {
-        // Get the Unit from the EmbeddedDataSpecification of the Property (if available)
+        // Get the Unit from the EmbeddedDataSpecification of the ConceptDescription of the Property (if available)
         unitSuffix(prop: any) {
-            // console.log('prop: ', prop);
-            if (prop.conceptDescription && prop.conceptDescription.embeddedDataSpecifications && prop.conceptDescription.embeddedDataSpecifications.length > 0 && prop.conceptDescription.embeddedDataSpecifications[0].dataSpecificationContent && prop.conceptDescription.embeddedDataSpecifications[0].dataSpecificationContent.unit) {
-                return prop.conceptDescription.embeddedDataSpecifications[0].dataSpecificationContent.unit;
+            if (prop.conceptDescriptions) {
+
+                for (let i = 0; i < prop.conceptDescriptions.length; i++) {

```

Comment:
Please use foreach here

Reply:
Even though "for" is generally faster, I agree that the readability of "foreach" and the avoidance of intermediate variables take precedence at this point.

But forEach has no built-in ability to break.
Whereas in the "for"-solution the loops interrupt by finding the first unit specification, the foreach loops are running through to the end.
Whereas in the "for"-solution the first occurance of a unit specification is used, the "foreach"-solution uses the last occurance of a unit specification.

Reply:
I agree with you. But in this case, performance is not a problem. This is mainly a cosmetic decision. This decision will also be part of the soon-released contributing guideline (and especially the coding conventions guideline).